go-selinux/selinux_linux.go

```diff
@@ -282,6 +282,23 @@ func readCon(fpath string) (string, error) {
 	return strings.Trim(retval, "\x00"), nil
 }
 
+// ClassIndex returns the int index for an object class in the loaded policy, or -1 and an error
+func ClassIndex(class string) (int, error) {
+	permpath := fmt.Sprintf("class/%s/index", class)
+	indexpath := filepath.Join(getSelinuxMountPoint(), permpath)
+
+	indexB, err := ioutil.ReadFile(indexpath)
+	if err != nil {
+		return -1, err
+	}
+	index, err := strconv.Atoi(string(indexB))
+	if err != nil {
+		return -1, err
+	}
+
```

Comment:
Perhaps we can cache the result in a map, so we won't have to reread it every time?

I'm not sure how often it is supposed to be called.

Reply:
One issue here would be in a long running process, the selinux policy could change and the results could change.  This should not be called enough where caching would matter.

Reply:
If we start doing a full avc implementation we will have to add caching but at that point we'll have a netlink socket so we'll know when the policy changes and can flush everything. Until then @rhatdan is correct, a long running process could possibly cache stale values